ext/session/session.c

```diff
@@ -1806,6 +1815,20 @@ PHP_FUNCTION(session_set_cookie_params)
 			goto cleanup;
 		}
 	}
+	if (same_site_enum) {
+		zval *case_name = zend_enum_fetch_case_name(same_site_enum);
+		samesite = zend_string_copy(Z_STR_P(case_name));
+
+		/* Verify that cookie is secure if using SameSite::None, see
+		 * https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite#samesitenone_requires_secure */
+		if (!secure && zend_string_equals_literal(samesite, "None")) {
```

Comment:
If this passes, this would be the second native enum. Leaving this link here for reference: https://github.com/php/php-src/pull/9679#discussion_r1045943444